src/main/webapp/app/exam/participate/exam-cover/exam-participation-cover.component.ts

```diff
@@ -75,16 +91,54 @@ export class ExamParticipationCoverComponent implements OnInit, OnDestroy {
     /**
      * check, whether exam has started yet and we therefore can enable the Start Exam Button
      */
-    enableStartButton() {
-        if (this.confirmed && this.exam && this.exam.startDate && moment(this.exam.startDate).isBefore(moment())) {
+    async enableStartButton() {
+        let fullname = '';
+        await this.accountService.identity().then((user) => {
+            fullname = user?.name ?? '';
+        });
+
+        if (this.fullname && this.fullname !== fullname) {
+            this.falseName = true;
+        } else {
+            this.falseName = false;
+        }
+
+        if (this.fullname && this.fullname === fullname && this.confirmed && this.exam && this.exam.startDate && moment(this.exam.startDate).isBefore(moment())) {
             return true;
         } else {
             return false;
         }
     }
 
     /**
-     * TODO: add session management, this function is bound to the start exam button
+     * Starts the exam and creates an exam session and stores the token in the browser
+     */
+    startExam() {
+        this.examSessionService.createExamSession(this.courseId, this.examId).subscribe((response) => {
+            const sessionToken = response.body?.sessionToken ?? '';
+            this.sessionStorage.store('ExamSessionToken', sessionToken);
+            const localSessionToken = this.sessionStorage.retrieve('ExamSessionToken');
+            console.log('Session Token', localSessionToken);
+
+            // TODO start exam
+        });
+    }
+
+    /**
+     * Submits the exam if user has valid token
      */
-    startExam() {}
+    submit() {
+        this.examSessionService.getCurrentExamSession(this.courseId, this.examId).subscribe((response) => {
+            const localSessionToken = this.sessionStorage.retrieve('ExamSessionToken');
```

Comment:
this.examSessionService.getCurrentExamSession should never be used. The check whether the session is still valid will happen on the server side